code/system/application/legacy/vanishingpoint/apps/LaneMarkingsVanishingPointDetectionModule.cpp

```diff
@@ -0,0 +1,283 @@
+#include <cstdlib>
+#include <iostream>
+
+#include <opendavinci/generated/odcore/data/image/SharedImage.h>
+#include <opendavinci/odcore/base/module/DataTriggeredConferenceClientModule.h>
+#include <opendavinci/odcore/base/Lock.h>
+#include <opendavinci/odcore/wrapper/SharedMemoryFactory.h>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+
+#include "VanishingPointDetection.h"
+#include "ConfigDeserialization.h"
+
+namespace lmvp {
+
+using odcore::base::module::DataTriggeredConferenceClientModule;
+using odcore::base::Lock;
+using odcore::data::Container;
+using odcore::data::image::SharedImage;
+using odcore::exceptions::ValueForKeyNotFoundException;
+using odcore::wrapper::SharedMemoryFactory;
+using odcore::wrapper::SharedMemory;
+
+void onMouse(int event, int x, int y, int flags, void * userdata);
+
+void onMouse(int event, int x, int y, int flags, void * userdata) {
+    LMVP_UNUSED(flags);
+    LMVP_UNUSED(userdata);
+
+    if(event == cv::EVENT_MOUSEMOVE) {
+        std::cout << "x: " << x << " y: " << y << std::endl;
+    }
+}
+
+class VanishingPointLowPass {
+public:
+    VanishingPointLowPass(float dampingFactor) : dampingFactor_(dampingFactor), lastVanishingPoint_() { }
```

Comment:
Thank you for the review. What is our rationale behind passing primitive data types by reference?